File: android_framework/services/src/main/java/github/tornaco/android/thanos/services/perf/PrefManagerStub.java
package github.tornaco.android.thanos.services.perf;

import github.tornaco.android.thanos.core.pref.IPrefManager;

public class PrefManagerStub extends IPrefManager.Stub {

    private final PreferenceManagerService provider;

    public PrefManagerStub(PreferenceManagerService provider) {
        this.provider = provider;
    }

    @Override
    public boolean putString(String name, String value) {
        return provider.putString(name, value);
    }

    @Override
    public String getString(String name, String def) {
        return provider.getString(name, def);
    }

    @Override
    public boolean putInt(String name, int value) {
        return provider.putInt(name, value);
    }

    @Override
    public int getInt(String name, int def) {
        return provider.getInt(name, def);
    }

    @Override
    public boolean getBoolean(String name, boolean def) {
        return provider.getBoolean(name, def);
    }

    @Override
    public boolean putBoolean(String name, boolean value) {
        return provider.putBoolean(name, value);
    }

    @Override
    public long getLong(String name, long def) {
        return provider.getLong(name, def);
    }

    @Override
    public boolean putLong(String name, long value) {
        return provider.putLong(name, value);
    }
}

File: android_framework/services/src/main/java/github/tornaco/android/thanos/services/perf/PrefManagerStub.kt
package github.tornaco.android.thanos.services.perf

import github.tornaco.android.thanos.core.pref.IPrefManager

class PrefManagerStub(private val provider: PreferenceManagerService) : IPrefManager.Stub() {

    override fun putInt(key: String, value: Int): Boolean {
        return provider.putInt(key, value)
    }

    override fun getInt(key: String, def: Int): Int {
        return provider.getInt(key, def)
    }

    override fun getBoolean(name: String, def: Boolean): Boolean {
        return provider.getBoolean(name, def)
    }

    override fun putBoolean(name: String, value: Boolean): Boolean {
        return provider.putBoolean(name, value)
    }

    override fun getLong(name: String, def: Long): Long {
        return provider.getLong(name, def)
    }

    override fun putLong(name: String, value: Long): Boolean {
        return provider.putLong(name, value)
    }

    override fun putString(name: String, value: String): Boolean {
        return provider.putString(name, value)
    }

    override fun getString(name: String, def: String): String? {
        return provider.getString(name, def)
    }
}
